config/admin-functions/lockdown.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -euo pipefail 
+
+update-initramfs -u
+
+# Remount / so it can't change while we're doing the veritysetup
+mount -o ro,remount /
+
+# Now do the dm-verity setup
+veritysetup format /dev/mapper/Vx--vg-root /dev/mapper/Vx--vg-hashes| tee "/tmp/verity.log"
+
+# Find the root hash and append it to our cmdline
+HASH="$(awk '/Root hash:/ { print $3 }' "/tmp/verity.log")"
+echo "$(cat /vx/admin/config/cmdline)${HASH}" > /tmp/cmdline
+
+# TODO: Make sure the output initramfs/kernel file names are correct!
+# Now package up ouer kernel, cmdline, etc
+objcopy \
+    --add-section .osrel="/usr/lib/os-release" --change-section-vma .osrel=0x20000 \
+    --add-section .cmdline="/tmp/cmdline" --change-section-vma .cmdline=0x30000 \
+    --add-section .splash="/vx/admin/config/logo.bmp" --change-section-vma .splash=0x40000 \
+    --add-section .linux="/boot/vmlinuz-5.10.0-11-amd64" --change-section-vma .linux=0x2000000 \
+    --add-section .initrd="/boot/initrd.img-5.10.0-11-amd64" --change-section-vma .initrd=0x3000000 \
+    "/usr/lib/systemd/boot/efi/linuxx64.efi.stub" "/tmp/linux.efi"
+
+# Sign the resulting binary
+# First make sure the drive is mounted
+mount /dev/sda /mnt
```

Comment:
what happens if this fails cause forgot to insert USB? I can live with not fixing this yet, but next quick fix would be to try this first and if that fails, error out and say "insert USB stick please"

Reply:
this is a good question. I will test and fix accordingly.

Reply:
So I tested with the USB unplugged, and with another USB plugged in that didn't have keys on it. When mount fails it just loops back into the admin interface. The only potentially sticky situation is if two USB block devices are plugged in at the same time, and the keys drive gets a name that isn't `sda` (like it could get `sdb`). Do we care?

Reply:
Nope, that's good enough for me.